### ISSUE TITLE
ld: recording.o: undefined reference to symbol 'curl_free'

### ISSUE THREAD

**Issue by amessina** [sipwise/rtpengine#251]
When building master@90242e36a313b8fba9e3eebb089d9f324c6817f6 on Fedora 23 x86_64, it looks like libcurl isn't linked in.

```
cc -g -Wall -pthread -fno-strict-aliasing -std=c99 `pkg-config --cflags glib-2.0` `pkg-config --cflags gthread-2.0` `pkg-config --cflags zlib` `pkg-config --cflags openssl` `pkg-config --cflags libevent_pthreads` `pcre-config --cflags` -I../kernel-module/ -D_GNU_SOURCE -DRTPENGINE_VERSION="\"4.4.1.1-2.git90242e3.fc23\"" -DRE_PLUGIN_DIR="\"/usr/lib/rtpengine\"" -O3 -o rtpengine main.o kernel.o poller.o aux.o control_tcp.o streambuf.o call.o control_udp.o redis.o bencode.o cookie_cache.o udp_listener.o control_ng.o sdp.o str.o stun.o rtcp.o crypto.o rtp.o call_interfaces.o dtls.o log.o cli.o graphite.o ice.o socket.o media_socket.o rtcp_xr.o homer.o recording.o -lm `pkg-config --libs glib-2.0` `pkg-config --libs gthread-2.0` `pkg-config --libs zlib` `pkg-config --libs libpcre` `pkg-config --libs libcrypto` `pkg-config --libs openssl` `pkg-config --libs libevent_pthreads` -lpcap `pcre-config --libs` `xmlrpc-c-config client --libs` -lhiredis
/usr/bin/ld: recording.o: undefined reference to symbol 'curl_free'
/usr/lib64/libcurl.so.4: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
```

**Comment by amessina** [comment]
The following change enables the build to succeed, though I'm not sure this would be your preferred fix.
```
 daemon/Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/daemon/Makefile b/daemon/Makefile
index 06d7c41..1ad4df3 100644
--- a/daemon/Makefile
+++ b/daemon/Makefile
@@ -54,6 +54,7 @@ LDFLAGS+=	-lpcap
 LDFLAGS+=	`pcre-config --libs`
 LDFLAGS+=	`xmlrpc-c-config client --libs`
 LDFLAGS+=	-lhiredis
+LDFLAGS+=       `pkg-config --libs libcurl`
 
 ifneq ($(DBG),yes)
   DPKG_BLDFLGS=	$(shell which dpkg-buildflags 2>/dev/null)
```

**Comment by rfuchs** [comment]
No, this is the correct fix, but I'm not too happy about depending on libcurl just for a string quoting function. I'd rather replace that function with another one.

**Comment by amessina** [comment]
@rfuchs I can understand that.  Does rtpengine actually require libcurl for anything else? https://github.com/sipwise/rtpengine#manual-compilation says that it's required, but I'm not sure what else it's used for -- it hasn't been linked in until recently.

**Comment by rfuchs** [comment]
It doesn't use it directly. Other libs may use it, xmlrpc for example. Leave your patch in for now - I'll remove the dependency in the very near future.

**Comment by amessina** [comment]
While the dependency may be removed in the future, this issue has been "fixed" by ba93a08d2468566b25c62ebb19062ee1e41c5d6c.
